adapter/integration-test/src/adapter-datasource.test.js

```diff
@@ -1,77 +1,125 @@
 const request = require('supertest')
 
 const {
-  ADAPTER_URL
+  ADAPTER_URL,
+  AMQP_URL
 } = require('./env')
 const { waitForServicesToBeReady } = require('./waitForServices')
+const {
+  connectAmqp,
+  consumeAmqpMsg
+} = require('./testHelper')
+
+const AMQP_EXCHANGE = 'ods_global'
+const AMQP_QUEUE = 'adapter-it-ds'
+const CONFIG_TOPIC = 'datasource.config.*'
+const CONFIG_CREATED_TOPIC = 'datasource.config.created'
+const CONFIG_DELETED_TOPIC = 'datasource.config.deleted'
+const CONFIG_UPDATED_TOPIC = 'datasource.config.updated'
 
 const TIMEOUT = 10000
+const publishedEvents = new Map() // routing key -> received msgs
+let amqpConnection
 
 describe('Datasource Configuration', () => {
   beforeAll(async () => {
     await waitForServicesToBeReady()
+
+    amqpConnection = await connectAmqp(AMQP_URL)
+
+    await consumeAmqpMsg(amqpConnection, AMQP_EXCHANGE, CONFIG_TOPIC, AMQP_QUEUE, publishedEvents)
   }, 60000)
 
+  afterAll(async () => {
+    await request(ADAPTER_URL)
+      .delete('/configs')
+      .send()
+    if (amqpConnection) {
+      await amqpConnection.close()
+    }
+  }, TIMEOUT)
+
   test('Should respond with stored datasource configs', async () => {
     const response = await request(ADAPTER_URL).get('/datasources')
     expect(response.status).toEqual(200)
     expect(response.type).toEqual('application/json')
 
-    expect(response.body).toEqual([])
+    expect(response.body).toBeInstanceOf(Array)
   }, TIMEOUT)
 
+  test('Should reject datasources with specified id [POST /datasources]', async () => {
```

Comment:
remove the URL because it can change. Rather say "when creating a datasource"

Reply:
That holds for the other tests as well, right? Do you think we should change all test names containing the endpoint?